.github/workflows/buf.yml

```diff
@@ -0,0 +1,20 @@
+on:
+  push:
+    branches:
+      - main
```

Comment:
I think you'd probably want two separate jobs - one that runs on `main` that does the push, and another that runs on every PR and does the other checks (lint, breaking). Maybe break this into two jobs?